diff --git a/app-deps/Android.bp b/app-deps/Android.bp
index b98d862..f04a840 100644
--- a/app-deps/Android.bp
+++ b/app-deps/Android.bp
@@ -12,7 +12,7 @@ java_defaults {
         "androidx.swiperefreshlayout_swiperefreshlayout",
         "androidx.cardview_cardview",
         "androidx.viewpager2_viewpager2",
-        "OpenEUICC_androidx.datastore_datastore-preferences",
+        "androidx.datastore_datastore-preferences",
         "OpenEUICC_com.journeyapps_zxing-android-embedded",
     ],
 }
\ No newline at end of file
